subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -99,14 +100,16 @@ public DefaultExecutionPlan(
         TaskDependencyResolver dependencyResolver,
         NodeValidator nodeValidator,
         ExecutionNodeAccessHierarchy outputHierarchy,
-        ExecutionNodeAccessHierarchy inputHierarchy
+        ExecutionNodeAccessHierarchy inputHierarchy,
+        ExecutionNodeAccessHierarchy destroyableHierarchy
```

Comment:
Nit: would be nice to get these in some logical order.

Reply:
Logical meaning input/output/destroys?

Reply:
That's a good one.

Reply:
Alphabetical would also work I think, but I like yours better.

Reply:
Done.